src/kafka-pubsub.ts

```diff
@@ -74,19 +74,22 @@ export class KafkaPubSub implements PubSubEngine {
    *
    * @param channel to use for internal routing, besides topic
    * @param payload event to send
+   * @param key the key of the event
    * @param headers optional kafkajs headers
    * @param sendOptions optional kafkajs producer.send options
    */
   public async publish(
     channel: string,
     payload: string | Buffer,
+    key?: string | Buffer,
```

Comment:
Let's add it last to avoid breaking compatibility with people already calling this function

Reply:
I have made the requested change